File: python/Problem66.py
# Problem: 66. Plus One
# URL: https://leetcode.com/problems/plus-one/
# Difficulty: Easy
# Tags: Array

from typing import List

class Solution:
  def plusOne(self, digits: List[int]) -> List[int]:
    result: List[int] = digits
    carry: int = 1
    i: int = len(digits) -1

    while carry != 0 and i >= 0:
      result[i] += carry
      carry = result[i] // 10
      result[i] %= 10
      i -= 1
    if carry != 0:
      result.insert(0, carry)
    return result

File: cpp/Problem66.cpp
/*
 * Problem: 66. Plus One
 * URL: https://leetcode.com/problems/plus-one/
 * Difficulty: Easy
 * Tags: Array
 */

#include <vector>

using namespace std;

class Solution {
public:
	vector<int> plusOne(vector<int>& digits) {
		int carry = 1;
		int i = digits.size() - 1;
		while (carry != 0 && i >= 0) {
			digits[i] += carry;
			carry = digits[i] / 10;
			digits[i] %= 10;
			i--;
		}
		if (carry != 0) {
			vector<int>::iterator it = digits.begin();
			digits.insert(it, carry);
		}
		return digits;
	}
};